### PR TITLE
innd: more conservative signal mask handling

### PR DESCRIPTION
**Pull request by ewxrjk**
Here we remove the assumption that the signal mask inherited at innd
startup was sensible. It's not an unreasonable assumption to have but a
user turns out to have a configuration that violates it.

Also added some more comments about the signal-masking logic.

### CODE REVIEW

**Comment by Julien-Elie** [comment]
Hi Richard, thanks for your PR.

I just have a newbie question (first time I deal with a PR...).
Well, I am a bit picky :-) but what should I do if I would like to homogenize the comments?
- line 53 with `*/" at the following line
- line 63 with a final dot at the end of the comment

Should you in fact amend your commit, or do a 2nd commit in the PR?  (though I believe it would be better to have only one; I see the "Squash and Merge" option but it seems it also adds a Merge commit after the change commit)
Or should I somehow merge it and amend it myself?  (I'll have a look at how to do it via GitHub; yet, it will mean that you would have to rebase your fork)

**Comment by ewxrjk** [comment]
Thanks for the feedback! When I review a PR, I leave comments on the lines I want changed and let the submitter update the PR. I'll add comments directly to demonstrate, then push an updated version.

File: lib/xsignal.c [hunk 1]
```diff
@@ -30,12 +48,21 @@ set_signal_handled(int signum, sig_handler_type sigfunc)
...
             signal_max = signum;
         }
         if (sigfunc != SIG_IGN && sigfunc != SIG_DFL) {
+            /* Block handled signals except during select, when
+             * we permit them.
+             */
             sigaddset(&signals_masked, signum);
+            sigdelset(&signals_unmasked, signum);
         } else {
+            /* Never block ignored/defaulted signals */
```
Comment (ewxrjk) [inline lib/xsignal.c:63]: Missing full stop.

File: lib/xsignal.c [hunk 2]
```diff
@@ -30,12 +48,21 @@ set_signal_handled(int signum, sig_handler_type sigfunc)
 {
     if (signal_masking) {
         if (signum > signal_max) {
+            /* We track the maximum handled signal so that
+             * we can efficiently reconfigure signals in
+             * xsignal_forked(). */
```
Comment (ewxrjk) [inline lib/xsignal.c:53]: Move `*/` to next line.

**Comment by ewxrjk** [comment]
I prefer single commits too, at least for things like this. (A complex change might make more sense as a series of commits, though.)

File: lib/xsignal.c (hunk 2 above)
Comment (Julien-Elie) [reply lib/xsignal.c:53]: Thanks Richard for the examples!

**Comment by ewxrjk** [comment]
> Here we remove the assumption that the signal mask inherited at innd
startup was sensible. It's not an unreasonable assumption to have but a
user turns out to have a configuration that violates it.

It turns out that the program that violated the assumption was innd itself.

The sequence of events is:
* unblock signals
* call `select`
* `ctliinnd xexec` sends a message to innd so `select` returns
* block signals
* exec the new `innd` (with signals blocked)

The fix remains the same, i.e. no further changes required; the user who reported the issue now reports that it is fixed.

**Pull request merged**